### ISSUE TITLE
Terraform 0.13 compatibility

### ISSUE THREAD

**Issue by jeffmccune** [openinfrastructure/terraform-google-multinic#38]

**Comment by jeffmccune** [comment]
See [Upgrading to Project Factory v10.0](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/docs/upgrading_to_project_factory_v10.0.md)

Need to fix this:

```
Error: Provider configuration not present

To work with
module.host.module.project.module.project-factory.null_resource.preconditions
its original provider configuration at
provider["registry.terraform.io/-/null"] is required, but it has been removed.
This occurs when a provider configuration is removed while objects created by
that provider still exist in the state. Re-add the provider configuration to
destroy
module.host.module.project.module.project-factory.null_resource.preconditions,
after which you can remove the provider configuration again.
```

Steps to reproduce:

 * Terraform v0.13.6
 * terragrunt version v0.27.3

```
cd ~/workspace/v6/platform/terraform/multinic/10_project
git checkout jm/38_tf13
cowboy terragrunt init -upgrade
cowboy terragrunt plan
```

**Comment by jeffmccune** [comment]
Upgrade process:

 1. For the project creation, apply project factory 9.2 and network 2.0 using TF 0.12.
 2. Apply the same with TF 0.13
 3. Update to project factor 10.x and network 3.x
 4. Apply the same with TF 0.13

Note this process avoids the above `Provider configuration not present` error.

**Comment by jeffmccune** [comment]
Released in 4.0.0.  Version 3.5.0 is the last version to support Terraform 0.12 and should be applied cleanly prior to updating to version 4.0.0 of this module.
